src/todolist.js

```diff
@@ -1,88 +1,93 @@
+  const [value, setValue] = useState("");
+  const [taskList, setTaskList] = useState(userData.tasklist);
+  const [date, setDate] = useState(new Date());
+  const [status, setStatus] = useState(isLoggedIn);
+
+  useEffect(() => {
+    setInterval(() => tick(), 1000);
+  }, []);
+  const tick = () => {
+    setDate(new Date());
+  };
 
-  handleChange = (event) => {
-    this.setState({ value: event.target.value });
+  const handleChange = (event) => {
+    setValue(event.target.value);
   };
 
   // Function to add task
-  onAddTask = (userInput) => {
+  const onAddTask = (taskValue) => {
+    console.log(new Date());
     // new todo object
-    const obj = {
-      name: userInput,
-      id: this.state.count + 1,
+    const task = {
+      name: taskValue,
+      id: new Date(),
     };
-    console.log("called");
-    let copy = this.state.todolist;
-    copy.push(obj);
 
     localStorage.setItem(
-      this.state.email,
-      JSON.stringify({ password: this.state.password, tasklist: copy })
+      email,
+      JSON.stringify({
+        password: userData.password,
+        tasklist: [...taskList, task],
+      })
     );
-    this.setState({
-      todolist: copy,
-      count: this.state.count + 1,
-    });
+    setTaskList([...taskList, task]);
   };
 
-  //Logout Funtion
-  logout = () => {};
+  const clearAll = () => {
+    setTaskList([]);
+    userData.tasklist = []
+    localStorage.setItem(email, JSON.stringify(userData));
+  };
+
+  //Logout Function
+  const logout = () => {
+    setStatus(false);
+  };
 
   //Function To delete Task
-  onDeleteTask = (itemId) => {
-    this.setState({
-      todolist: [...this.state.todolist].filter((id) => id.id !== itemId),
-    });
-    localStorage.setItem(
-      this.state.email,
-      JSON.stringify({
-        password: this.state.password,
-        tasklist: this.state.todolist,
-      })
-    );
+  const onDeleteTask = (itemId) => {
+    setTaskList([...taskList].filter((id) => id.id !== itemId));
+    userData.tasklist = [...taskList].filter((id) => id.id !== itemId);
+    localStorage.setItem(email, JSON.stringify(userData));
   };
 
-  // Rendering Todolist Component
-  render() {
-    const mylist = this.state.todolist.map((todo) => (
-      <li key={todo.id}>
-        {todo.name}
-        <button onClick={() => this.onDeleteTask(todo.id)}>Remove</button>
-      </li>
-    ));
-    return (
-      <div className="App">
-        {this.props.isLoggedIn ? (
-          <>
-            <h1>TodoList</h1>
-            <ul>{mylist}</ul>
-            <input onChange={this.handleChange}></input>
-            <button onClick={() => this.onAddTask(this.state.value)}>
-              {" "}
-              Add Task{" "}
-            </button>
-            <button onClick={() => this.logOut()}>Logout</button>
-          </>
-        ) : (
-          <h1>Please Log in</h1>
-        )}
-        ;
-      </div>
-    );
-  }
+  const Task = ({ task, onDeleteTask }) => (
```

Comment:
this is a component , define it outside this component and in seprate file , cleanup and modularise the code